### PR TITLE
Start documentation of sum_to_zero_vector

### LINKED ISSUE #804: Document sum_to_zero_vector
#### Summary:
Implemented in https://github.com/stan-dev/stanc3/pull/1443, these should be documented before the next release.

A lot of material may be adapted from the existing documentation [here](https://mc-stan.org/docs/stan-users-guide/regression.html#parameterizing-centered-vectors)

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submission Checklist

- [x] Builds locally
- [x] New functions marked with `` <<{ since VERSION }>>``
- [x] Declare copyright holder and open-source license: see below

#### Summary

Closes #804.

I think I need some help from @spinkney @bob-carpenter filling out the documentation of the specific transform, see the two "TODO:" lines in `transforms.qmd`

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):
Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC BY-ND 4.0 (https://creativecommons.org/licenses/by-nd/4.0/)

### CODE REVIEW

**Comment by spinkney** [comment]
I'll try and take a look this week

**Comment by bob-carpenter** [comment]
I can take a shot at this, @spinkney.  I think the definition is clear enough.

**Comment by bob-carpenter** [comment]
I have most of this written up and just have to double check.  I should be able to push the doc tomorrow (Wednesday).  I'm just transating from the code.  It's really nice that there's no Jacobian adjustment.

**Comment by bob-carpenter** [comment]
I wrote out the math for the transform.  It'd be great if @spinkney could review, though I'm pretty sure it at least matches the code as is (perhaps modulo a +1 or -1 on indexing, but I triple-checked all that, too).

**Comment by WardBrian** [comment]
@spinkney do you think you will have a chance to look at this?

**Comment by spinkney** [comment]
> @spinkney do you think you will have a chance to look at this?

When is the deadline?

**Comment by WardBrian** [comment]
The 11th, unless we end up delaying the release again

**Comment by spinkney** [comment]
I will either finish this before 11/9 or let you know I won't be able to make the deadline.

**Comment by bob-carpenter** [comment]
It's very minimal and it's all written---just needs someone to verify it's OK.

**Comment by spinkney** [comment]
> It's very minimal and it's all written---just needs someone to verify it's OK.

I'm on vacation and coming back tomorrow. I'll take a look then.

**Comment by spinkney** [comment]
I'll have time to look at it this weekend/monday

**Comment by spinkney** [comment]
looking at this later this morning. Sorry about the delay!

**Review by spinkney — CHANGES_REQUESTED** [review]
Added a bunch of details. We may even want to write out the Householder stuff. Adrian wrote about it here https://github.com/pyro-ppl/numpyro/pull/1751#issuecomment-1980569811. I took most of the derivation from something I posted on discourse at https://discourse.mc-stan.org/t/new-stan-data-type-zero-sum-vector/26215/11?u=spinkney.

File: src/reference-manual/transforms.qmd [hunk 1]
```diff
@@ -462,6 +462,83 @@ p_Y(y)
...
+Vectors that are constrained to sum to zero are useful for, among
+other things, additive varying effects, such as varying slopes or
+intercepts in a regression model (e.g., for income deciles).
+
+A zero sum $K$-vector $x \in \mathbb{R}^K$ satisfies the constraint
+$$
+\sum_{k=1}^K x_k = 0.
+$$
+
+For the transform, Stan uses the first part of an isometric log ratio
```
[the remark this answers is not in the document]
Comment (bob-carpenter) [reply src/reference-manual/transforms.qmd:477]: I wouldn't say "just a reference manual".  We want to get the details of the transformation we're using right.  This isn't a research paper, though, so we don't need to say why we're using this, though we can link to a paper or to Adrian's discussion online.  External links are challenging to maintain, so use sparingly.

We can mention the link to Householder, but I wouldn't go through this whole discussion in the transforms chapter of the Stan reference manual.

[the remark this answers is not in the document]
Comment (bob-carpenter) [reply src/reference-manual/transforms.qmd:477]: I'm not sure how to update.  Do you want to take a stab at just updating it?  If not, if you can write here what you think I should add, I'll add it.

**Comment by bob-carpenter** [comment]
I added a short note and citation in the reference manual and then just used your text in full in the user's guide.  I got rid of the simplex suggestion we had, but left the soft-centering.

**Review by spinkney — APPROVED** [review]
Just the one double space and the rest looks good!

File: src/reference-manual/types.qmd [hunk 2]
```diff
@@ -692,6 +693,26 @@ unit vectors, this is only done up to a statically specified accuracy
...
+While the underlying vector has only $N - 1$ degrees of freedom,
+zero sum vectors are declared with their full dimensionality.
+For instance, `beta` is declared to be a zero-sum $5$-vector (4 DoF) by
+
+```stan
+sum_to_zero_vector[5] beta;
+```
+
+Zero sum vectors are implemented as vectors and may be assigned to other
+vectors and vice-versa.  Zero sum  vector variables, like other constrained
```
[the remark this answers is not in the document]
Comment (bob-carpenter) [reply src/reference-manual/types.qmd]: Fixed.  The spaces don't matter for rendering.

**Comment by bob-carpenter** [comment]
@spinkney I think this is then ready for final review.  I removed the extra space.

File: src/reference-manual/transforms.qmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (lingium) [reply src/reference-manual/transforms.qmd:477]: I think your description of  "zero correlation" might be wrong. Think about a vector that sum to zero, if one element becomes larger, then other elements `must` be smaller, right? So as long as they sum to zero, they are all negativly correlated.

**Comment by spinkney** [comment]
Absolutely! I don't know why I keep saying correlation. The correct thing
here is that the transform we have results in equal variances. We need to
add in a zero_sum_normal which results in the zero sum elements having mean
0 and stdev equal to the user specified value which will be the same as
N(0, val). I'll put a fix in for the docs later today, unless someone else
gets to it before.

**Comment by bob-carpenter** [comment]
Also, when discussing correlation, there's the issue of whether we're talking about the `N - 1` unconstrained parameters or the `N` constrained parameters.

**Comment by lingium** [comment]
yeah, unconstrained parameters are independent, and constrained parameters are negatively (but equally, guess @spinkney also would like to mentin this) correlated

**Pull request merged**